tests/test_es_doc_creator.py

```diff
@@ -0,0 +1,1097 @@
+    )
+    assert (
+        document_creator.try_state_as_boolean(State("domain.entity_id", "on")) is True
+    )
+    assert (
+        document_creator.try_state_as_boolean(State("domain.entity_id", "off")) is True
+    )
+    assert (
+        document_creator.try_state_as_boolean(State("domain.entity_id", "1")) is False
+    )
+    assert (
+        document_creator.try_state_as_boolean(State("domain.entity_id", "0")) is False
+    )
+    assert (
+        document_creator.try_state_as_boolean(State("domain.entity_id", "1.0")) is False
+    )
+
+    assert (
+        document_creator.try_state_as_boolean(
+            State("domain.entity_id", MOCK_NOON_APRIL_12TH_2023)
+        )
+        is False
+    )
+
+
+@pytest.mark.asyncio
+async def test_state_as_datetime(
+    hass: HomeAssistant, document_creator: DocumentCreator
+):
+    """Test state to datetime conversion."""
+
+    assert document_creator.state_as_datetime(
+        State("domain.entity_id", MOCK_NOON_APRIL_12TH_2023)
+    ) == dt_util.parse_datetime(MOCK_NOON_APRIL_12TH_2023)
+
+    with pytest.raises(ValueError):
+        document_creator.state_as_datetime(State("domain.entity_id", "tomato"))
+
+    with pytest.raises(ValueError):
+        document_creator.state_as_datetime(State("domain.entity_id", "1"))
+
+    with pytest.raises(ValueError):
+        document_creator.state_as_datetime(State("domain.entity_id", "0"))
+
+    with pytest.raises(ValueError):
+        document_creator.state_as_datetime(State("domain.entity_id", "on"))
+
+    with pytest.raises(ValueError):
+        document_creator.state_as_datetime(State("domain.entity_id", "off"))
+
+
+async def test_try_state_as_datetime(
+    hass: HomeAssistant, document_creator: DocumentCreator
+):
+    """Test state to datetime conversion."""
+
+    assert (
+        document_creator.try_state_as_datetime(State("domain.entity_id", "tomato"))
+        is False
+    )
+    assert (
+        document_creator.try_state_as_datetime(State("domain.entity_id", "1")) is False
+    )
+    assert (
+        document_creator.try_state_as_datetime(State("domain.entity_id", "0")) is False
+    )
+    assert (
+        document_creator.try_state_as_datetime(State("domain.entity_id", "on")) is False
+    )
+    assert (
+        document_creator.try_state_as_datetime(State("domain.entity_id", "off"))
+        is False
+    )
+    assert (
+        document_creator.try_state_as_datetime(
+            State("domain.entity_id", "2023-04-12T12:00:00Z")
+        )
+        is True
+    )
+
+
+async def test_state_to_entity_details(hass: HomeAssistant):
+    """Test entity details creation."""
+    es_url = "http://localhost:9200"
+
+    mock_entry = MockConfigEntry(
+        unique_id="test_entity_details",
+        domain=DOMAIN,
+        version=3,
+        data=build_full_config({"url": es_url, CONF_INDEX_MODE: INDEX_MODE_LEGACY}),
+        title="ES Config",
+    )
+
+    await _convert_mock_config_to_config(hass, mock_entry)
```

Comment:
Is this still needed, or can we transition to use `get_merged_config(mock_entry)`?

Reply:
I think the config needs to be actually inserted to link a device entry against it and so get_merged_config won't suffice. In the end the way I was using this didn't differ from _setup_config_entry in a meaninful way so I'll switch it out for that function